apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -50,6 +52,56 @@
     this.logger.debug('Clock updated', now);
   };
 
+  LockScreenClockWidget.prototype.updateAlarm =
+	function() {
+    this.fetchAlarmData()
+    .then((alarmData) => {
+
+      // if there is no alarm
+      if(!alarmData.hour && !alarmData.minute) {
+        this.resources.elements.alarm.classList.add('no-alarms');
+        return;
+      }
+
+      var type = 'AM',
+          hour = parseInt(alarmData.hour, 10);
+
+      // Decide wheter to use AM or PM
+      if(hour > 12) {
+        type = 'PM';
+        hour = hour - 12;
+      }
+
+      var fullAlarmTime = hour + ':' + alarmData.minute + type;
+      this.resources.elements.alarmtime.textContent = fullAlarmTime;
+      this.resources.elements.alarm.classList.remove('no-alarms');
+      this.resources.elements.alarm.classList.add('has-alarm');
+      this.logger.debug('Alarm updated to :', fullAlarmTime);
+    }).catch(() => {
```

Comment:
Don't use `catch` as control flow... As I mentioned in the previous commits.
It's really improper because:

1. You will ignore the real error, since you ignore the argument here.
2. Even you give it a check, and throw the real exception, it's still confused since you fork the semantic meaning of an exception to two types: one is your **expected** result and thus it's **obvious not an exception**, and another is the **real exception**.
3. What I've suggested is to pass an empty or specialized object for the **no alarm data data**, which is what you're checking here. You can do this in the `fetchAlarmData` method as I comment below.